internal/process/defunct_processes.go

```diff
@@ -0,0 +1,90 @@
+package process
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// ProcessFS is the process file system.
+const ProcessFS = "/proc"
+
+// Stat represents status information of a process from /proc/[pid]/stat.
+type Stat struct {
+	// Comm is the command name (usually the executable filename).
+	Comm string
+
+	// State is the state of the process.
+	State string
+}
+
+// DefunctProcesses returns the number of zombie processes in the node.
+func DefunctProcesses() (defunctCount uint, err error) {
+	directories, err := os.Open(ProcessFS)
+	if err != nil {
+		return 0, err
+	}
+	defer directories.Close()
+
+	names, err := directories.Readdirnames(-1)
+	if err != nil {
+		return 0, err
+	}
+
+	for _, name := range names {
+		// Processes have numeric names. If the name does not start with a number, skip.
+		if name[0] < '0' || name[0] > '9' {
+			continue
+		}
```

Comment:
this reads as duplicate of ParseInt. is there any reason to not just call ParseInt here?

Reply:
@haircommander Essentially yeah, `ParsInt()` would take care of this case, but I was trying to filter out the directory names that don't start with a number and hence, avoid calling `ParseInt()` on every directory name to reduce the number of function calls. Could get rid of it if the overhead of too many function calls is not negligible :)

Reply:
@saschagrunert what do you think?

Reply:
I assume it wouldn’t be faster with this check, so yes let’s just use ParseInt